test/test_sign_command.rb

```diff
@@ -0,0 +1,44 @@
+require 'helper'
+require "rubygems/commands/sign_command"
+
+class TestSignCommand < Gem::TestCase
+  include SigstoreAuthHelper
+  include FulcioHelper
+  include RekorHelper
+
+  def setup
+    super
+
+    @gem_path = gem_path("hello-world.gem")
+    @cmd = Gem::Commands::SignCommand.new
+  end
+
+  def test_sign
+    @cmd.options[:args] = [@gem_path]
+
+    stub_sigstore_auth_get_openid_config
+    stub_sigstore_auth_create_token
+    stub_sigstore_auth_get_keys
+    stub_fulcio_create_signing_cert
+    stub_rekor_create_log_entry(gem_digest(@gem_path))
```

Comment:
I would suggest moving the happy path into the setup method, so that other tests only change what is relevant to them.